final_task/rss_reader/classes/news_class.py

```diff
@@ -40,16 +46,37 @@ def create_string_of_links(self):
         return string_repr_of_links
 
 
+    def print_feed_title(self):
+        """Print feed title"""
+        if self.command_line_args.colorize:
+            cprint(self.feed_title, 'magenta')
+        else:
+            print(self.feed_title)
+
+
     def print_news(self):
         """Print to stdout title, date, link, text
         and links from text in particular sequence."""
-        print('-----------------------------------------------------------')
-        print('Title: '+self.title)
-        print('Date: '+self.date)
-        print('Link: '+self.link, end='\n\n')
-        print(self.text, end='\n\n')
-        links = self.create_string_of_links()
-        if links:
-            print('Links:')
-            print(links)
-        print('-----------------------------------------------------------')
+        if self.command_line_args.colorize:
+            cprint('-----------------------------------------------------------', 'magenta')
+            cprint('Title: '+self.title, 'magenta')
+            cprint('Date: '+self.date, 'magenta')
+            cprint('Link: '+self.link, 'magenta', end='\n\n')
+            cprint(self.text, 'green', end='\n\n')
+            links = self.create_string_of_links()
+            if links:
+                cprint('Links:', 'blue')
+                cprint(links, 'blue')
+            cprint('-----------------------------------------------------------', 'magenta')
+
+        else:
+            print('-----------------------------------------------------------')
+            print('Title: '+self.title)
+            print('Date: '+self.date)
+            print('Link: '+self.link, end='\n\n')
+            print(self.text, end='\n\n')
+            links = self.create_string_of_links()
+            if links:
+                print('Links:')
+                print(links)
+            print('-----------------------------------------------------------')
```

Comment:
You have almost complete code duplication in these two statements. Try to rethink this logic. For example you could use a decorator here

Reply:
Fixed